### PR TITLE
Implement logout functionality

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Addresses issues:
* https://github.com/solid/oidc-op/issues/6
* https://github.com/solid/oidc-auth-manager/issues/20
* https://github.com/solid/node-solid-server/issues/675

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
This is a first pass at implementing a better Logout functionality on the server side. 
What it currently does:
- Calls the host server's `logout()` (to clear any sessions or whatever)
- Validates `post_logout_redirect_uri` per the spec, and 302 redirects to it if present (along with the `state`)
- Requires `id_token_hint` to be present if `post_logout_redirect_uri` is given (since the token hint is the only way to validate the pre-registered redirect uri).
- Responds with a 204 No Content if no redirect uri is given

Things to do in the next iteration (and things to discuss):
- Doesn't prompt the user for consent to also log out of the home pod
- If no redirect uri is given by the RP client, should we redirect to a pod's default `/goodbye` uri or something? (Instead of just a 204 No Content)
- Possibly consider as a special case post-logout redirect uris if they're on the user's home pod?

**Comment by dmitrizagidulin** [comment]
Ok, changed it so that if no `post_logout_redirect_uri` is provided by the RP, it looks in the host/OP's defaults (to redirect to `/goodbye`, for example).

PR is mostly ready for review, needs a few more tests though.

File: test/handlers/RPInitiatedLogoutRequestSpec.js [hunk 1]
```diff
@@ -59,41 +71,34 @@ describe('RPInitiatedLogoutRequest', () => {
     it('should validate that `post_logout_redirect_uri` has been registered')
   })
 
-  describe('redirectOrRespond()', () => {
-    it('should redirect to RP if logout uri provided', done => {
+  describe('redirectToGoodbye()', () => {
+    it('should redirect to RP if logout uri provided', () => {
       let res = HttpMocks.createResponse()
       let req = HttpMocks.createRequest({
         method: 'GET',
```
Comment (dmitrizagidulin) [inline test/handlers/RPInitiatedLogoutRequestSpec.js]: Tests need to be changed to POST

**Comment by kjetilk** [comment]
Hi! We're having the meeting with @kidehen and we found that this PR should fix the logout problem without merging https://github.com/solid/solid-auth-client/pull/46 (because that PR fixes the problem in the wrong place). 

The problem has been that we haven't felt quite up to the task of reviewing this, so we figured that OpenLink would be better qualified to review it. So, @cblakeley could you please test and review this PR and report back? Make sure that you do not have https://github.com/solid/solid-auth-client/pull/46 running when you test this, since that would make it appear to log out, but it would be the wrong way.

**Comment by dmitrizagidulin** [comment]
@kjetilk Thanks for the heads up. I've resolved the conflicts, should be fine now.

(Part of the confusion about the timeouts has to do with -- so this PR moves all the individual timeouts into a single Mocha-wide timeout in `mocha.opts`.)

**Pull request merged**